fineract-provider/src/main/java/org/apache/fineract/cob/loan/AbstractLoanItemProcessor.java

```diff
@@ -40,17 +42,25 @@ public abstract class AbstractLoanItemProcessor implements ItemProcessor<Loan, L
 
     @Setter(AccessLevel.PROTECTED)
     private ExecutionContext executionContext;
+    private LocalDate businessDate;
 
     @Override
     public Loan process(@NotNull Loan item) throws Exception {
         TreeMap<Long, String> businessStepMap = (TreeMap<Long, String>) executionContext.get(LoanCOBConstant.BUSINESS_STEP_MAP);
 
-        return cobBusinessStepService.run(businessStepMap, item);
+        Loan alreadyProcessedLoan = cobBusinessStepService.run(businessStepMap, item);
+        alreadyProcessedLoan.setLastClosedBusinessDate(businessDate);
+        return alreadyProcessedLoan;
     }
 
     @AfterStep
     public ExitStatus afterStep(@NotNull StepExecution stepExecution) {
         return ExitStatus.COMPLETED;
     }
 
+    protected void setBusinessDate(StepExecution stepExecution) {
```

Comment:
You sure we need this? We already have the executioncontext within this class and using it in the process method. Why don't we do it similarly?

Reply:
we need this when tha business date is in different ExecutionContext than the other parameters we use (i.e. LoanCOB)